goalgorilla/open_social
# DS-4122 - Move front-end libraries to composer.json

## Description

This PR has a lot of changes, sorry for that. However we already split it up a little bit, see https://github.com/goalgorilla/open_social/pull/519.

The idea is to remove all of the front-end libraries from the codebase and require them via a package manager. This means we removed a lot of code! And @MaikelGG rewrote the search script to make it work.

See also PR https://github.com/goalgorilla/open_social/pull/530 and https://github.com/goalgorilla/drupal_social/pull/320.

## Important

Make sure you followed instructions at https://github.com/goalgorilla/open_social/pull/530. 

## How to test

If https://github.com/goalgorilla/drupal_social/pull/320 is merged, it makes it easier for you.
- [x] Install this new version, see that the assets are now installed in html/libraries
- [x] Also install it via the makefile, it should also be in html/libraries
- [x] Make sure **all** tests pass, be aware that the travis builds will fail unless https://github.com/goalgorilla/drupal_social/pull/320 is merged

Please note that we did not really update any library versions, except for Waves. Morris and Raphael didn't change in version either and are only used by the analytics module, so hard to test. Now test the different libraries by:
- [x] Waves library by clicking on buttons (should cause a wave in the button)
- [x] Timepicker library by creating an event, you can pick the time via this tool
- [x] Click through the site for the Bootstrap library, create a group, join an event, check tabs on user profile page, etc.
